diff -Nur gmail-notify-1.6.1/GmailConfig.py gmail-notify-1.6.1.new/GmailConfig.py
--- gmail-notify-1.6.1/GmailConfig.py	2005-09-25 19:12:11.000000000 +0200
+++ gmail-notify-1.6.1.new/GmailConfig.py	2006-04-09 00:06:01.000000000 +0200
@@ -237,7 +237,10 @@
 				self.config.remove_option( "options", "gmailusername" )
 				self.config.remove_option( "options", "gmailpassword" )
 
-			self.config.write( open( self.loadedConfig, 'w' ) )
+			fd = open( self.loadedConfig, 'w' )
+			os.chmod( self.loadedConfig, 0600)
+			self.config.write( fd )
+			fd.close()
 			gtk.main_quit()
 			self.hide()
 		else:
