Description: Drop in-program git version check
Author: Axel Beckert <abe@debian.org>
Debian-Bug: 692884

Index: guilt/guilt
===================================================================
--- guilt.orig/guilt	2013-06-17 18:47:39.000000000 +0200
+++ guilt/guilt	2013-06-17 22:57:35.000000000 +0200
@@ -26,17 +26,6 @@
 . "$(git --exec-path)/git-sh-setup"
 
 #
-# Git version check
-#
-gitver=`git --version | cut -d' ' -f3 | sed -e 's/^debian\.//'`
-case "$gitver" in
-	1.5.*)	;; # git config
-	1.6.*)	;; # git config
-	1.7.*)	;; # git config
-	*)	die "Unsupported version of git ($gitver)" ;;
-esac
-
-#
 # Shell library
 #
 usage()
