Description: Fix "guilt help" man page path
Author: Axel Beckert <abe@debian.org>
Debian-Bug: 712620

Index: guilt/guilt-help
===================================================================
--- guilt.orig/guilt-help	2013-06-17 18:47:39.000000000 +0200
+++ guilt/guilt-help	2013-06-18 01:13:58.000000000 +0200
@@ -34,7 +34,7 @@
 		;;
 esac
 
-MANDIR=`dirname $0`/../man
+MANDIR=`dirname $0`/../share/man
 MANDIR=`(cd "$MANDIR"; pwd)`
 exec man -M "$MANDIR" "$page"
 
