diff --git a/gcode/autoproxy/chrome/content/requests.js b/gcode/autoproxy/chrome/content/requests.js
index 32d54e5..29083c5 100644
--- a/gcode/autoproxy/chrome/content/requests.js
+++ b/gcode/autoproxy/chrome/content/requests.js
@@ -460,7 +460,7 @@ RequestEntry.prototype =
 function getWeakReference(/**nsISupports*/ node) /**nsIWeakReference*/
 {
   if (node instanceof Ci.nsISupportsWeakReference)
-    return node.GetWeakReference();
+    return Cu.getWeakReference(node);
   else
     return null;
 }
